cql3/statements/create_keyspace_statement.cc

```diff
@@ -76,6 +77,40 @@ void create_keyspace_statement::validate(query_processor& qp, const service::cli
     } catch (const std::runtime_error& e) {
         throw exceptions::invalid_request_exception(e.what());
     }
+
+    const auto& tm = *qp.proxy().get_token_metadata_ptr();
```

Comment:
The protocol for token_metadata_ptr requires that you hold it for the duration of execution, you can't just dereference it.


Since the function doesn't return a future, you can't yield and cause a use after free, but it's better not to rely on such things and capture the pointer for the duration.